python/rascaline/utils.py

```diff
@@ -20,3 +22,225 @@ def _call_with_growing_buffer(callback, initial=1024):
+            "soap_type",
+            "compute_gradients",
+        ]
+        if any([d in hyperparameters for d in deprecated_params]):
+            warnings.warn(
+                "{} are not required parameters in the rascaline software infrastructure".format(
+                    ",".join(
+                        [f"`{d}`" for d in deprecated_params if d in hyperparameters]
+                    )
+                )
+            )
+
+        not_supported = [
+            "coefficient_subselection",
+            "covariant_lambda",
+            "gaussian_sigma_type",
+            "inversion_symmetry",
+            "normalize",
+            "optimization_args",
+            "optimization",
+        ]
+        if any([d in hyperparameters for d in not_supported]):
+            warnings.warn(
+                "{} are not currently supported in rascaline".format(
+                    ",".join([f"`{d}`" for d in not_supported if d in hyperparameters])
+                )
+            )
+
+        return {k: v for k,v in new_hypers.items() if v is not None}
+    elif mode == "dscribe":
+        anticipated_hypers = [
+            "rcut",
+            "nmax",
+            "lmax",
+            "species",
+            "sigma",
+            "rbf",
+            "periodic",
+            "crossover",
+            "average",
+            "sparse",
+            "dtype",
+        ]
+
+        if any([key not in anticipated_hypers for key in hyperparameters]):
+            raise ValueError(
+                "I do not know what to do with the following hyperparameter entries:\n\t".format(
+                    "\n\t".join(
+                        [
+                            key
+                            for key in hyperparameters
+                            if key not in anticipated_hypers
+                        ]
+                    )
+                )
+            )
+
+        new_hypers["atomic_gaussian_width"] = hyperparameters.pop("sigma", None)
+        new_hypers["max_angular"] = hyperparameters.pop("lmax", None)
+        new_hypers["max_radial"] = hyperparameters.pop("nmax", None)
+        new_hypers["cutoff"] = hyperparameters.pop("rcut", None)
+
+        if  'rbf' in hyperparameters:
+            new_hypers["radial_basis"] = {hyperparameters.pop("rbf").title(): {}}
+            if new_hypers["radial_basis"] != "Gto":
+                warnings.warn("WARNING: rascaline currently only supports a Gto basis.")
+
+        deprecated_params = ["average", "sparse", "dtype"]
+        if any([d in hyperparameters for d in deprecated_params]):
+            warnings.warn(
+                "{} are not required parameters in the rascaline software infrastructure".format(
+                    ",".join(
+                        [f"`{d}`" for d in deprecated_params if d in hyperparameters]
+                    )
+                )
+            )
+
+        not_supported = [
+            "periodic",
+            "crossover",
```

Comment:
`crossover=True` is what rascaline does, `crossover=False` could be achieved once #134 is merged with a `selected_keys` parameter to the compute function